spec-tests/src/test/java/org/candlepin/spec/imports/ImportWarningTest.java

```diff
@@ -0,0 +1,247 @@
+        expiredEnt.setPool(pool);
+
+        File modifiedExport = File.createTempFile(export.file().getName().replace(".zip", ""), ".zip");
+        modifiedExport.deleteOnExit();
+
+        // Create a copy of the export to replace the existing entitlement with the expired one.
+        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(modifiedExport))) {
+            try (ZipFile exportZip = new ZipFile(exportFile)) {
+                exportZip.stream().forEach(entry -> {
+                    if (entry.getName().equals("consumer_export.zip")) {
+                        File temp = null;
+                        try {
+                            temp = File.createTempFile("consumer_export", ".zip");
+                            temp.deleteOnExit();
+                        }
+                        catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+
+                        // Create a copy of the consumer_export.zip with the expired entitlement
+                        String entToModify = ENTITILEMENTS_PATH + expiredEnt.getId() + ".json";
+                        try (ZipOutputStream zosTemp = new ZipOutputStream(new FileOutputStream(temp))) {
+                            ZipFile archive = ExportUtil.getExportArchive(export.file());
+                            archive.stream().forEach(archiveEntry -> {
+                                if (archiveEntry.getName().equals(entToModify)) {
+                                    // Change the entitlement to the expired entitlement
+                                    try (InputStream istream = archive.getInputStream(archiveEntry)) {
+                                        byte[] entBytes = getEntitlementJson(expiredEnt);
+                                        zosTemp.putNextEntry(new ZipEntry(entToModify));
+                                        zosTemp.write(entBytes, 0, entBytes.length);
+                                    }
+                                    catch (IOException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                }
+                                else {
+                                    // Copy over entries we are not modifying
+                                    transfer(archive, archiveEntry, zosTemp);
+                                }
+                            });
+                        }
+                        catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+
+                        // Copy over the modified consumer_export.zip
+                        transfer(entry, temp, zos);
+                    }
+                    else {
+                        // Copy the signature file
+                        transfer(exportZip, entry, zos);
+                    }
+                });
+            }
+        }
```

Comment:
Could we extract this into some methods? Just to be more readable for future.

Reply:
That is a good idea! I created `copyConsumerExportAndReplaceEnt` helper method and moved the logic that creates a copy of the consumer_export.zip file in the Export and replaced the entitlement in that copy.